src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSRequest.java

```diff
@@ -74,6 +75,8 @@ private HttpRequest getApiRequest(Map<String, Object> postData, GenericUrl endpo
                 // user agent string and it will grow big enough to cause header overflow
                 // errors.
                 .setHeaders(this.httpHeaders.clone())
+                .setReadTimeout(this.timeout)
+                .setConnectTimeout(this.timeout)
```

Comment:
Just so I understand, does this mean it might wait up to 40 seconds to make a request?

Reply:
Yes, in the worst case where it takes a while to get an ACK back from the TSP address _and_ takes a while to get the response from the KMS.